cpp/src/arrow/util/decimal.h

```diff
@@ -129,6 +129,20 @@ class ARROW_EXPORT Decimal128 {
   /// \brief Convert Decimal128 from one scale to another
   Status Rescale(int32_t original_scale, int32_t new_scale, Decimal128* out) const;
 
+  template <typename T, typename = typename EnableIfIsOneOf<T, int32_t, int64_t>>
+  Status to(T* out) const {
```

Comment:
IMHO it would be clearer, more explicit, and more consistent with the other methods if this were called `ToInteger`.